zerver/lib/utils.py

```diff
@@ -204,3 +206,27 @@ def check_subdomain(realm_subdomain, user_subdomain):
         if realm_subdomain != user_subdomain:
             return False
     return True
+
+def validate_domain(domain):
+        # type: (Text) -> Tuple[bool, str]
+        error_strings = {
+            'empty': _("Domain can't be empty."),
+            'too short': _("Domain needs to have length 3 or greater."),
+            'missing dot': _("Domain must have atleast one dot(.)."),
+            'extremal dash': _("Subdomain cannot start or end with a '-'."),
+            'extremal dot': _("Domain cannot start or end with a dot(.)."),
+            'bad character': _("Domain can only have lowercase letters, numbers, and '-'s.")}
+        if not domain:
+            return False, error_strings['empty']
+        if len(domain) < 3:
+            return False, error_strings['too short']
+        if '.' not in domain:
+            return False, error_strings['missing dot']
```

Comment:
ah, I see where you got the `error_strings` pattern :). In `RegistrationForm` we needed it because we needed to sometimes use "subdomain" and sometimes use "short name". Here we are always using the same thing, so the error strings can be placed in-line. E.g.
```
if '.' not in domain:
    raise ValidationError(_("Domain cannot start or end with a dot (.)"))
```
(For the dot one I'd also remove the sentence ending period, and write it like above)